CMakeLists.txt

```diff
@@ -264,6 +264,9 @@ zephyr_system_include_directories(${NOSTDINC})
 # Force an error when things like SYS_INIT(foo, ...) occur with a missing header.
 zephyr_cc_option(-Werror=implicit-int)
 
+# Prohibit void pointer arithmetic. Illegal in C99
+zephyr_cc_option(-Wpointer-arith)
```

Comment:
-Werror=pointer-arith ?

Unless there is a good reason to allow pointer arithmetic during development ...

Reply:
Too aggressive? I did consider it but may lead to widespread breakage for some boards accustomed to GCC.

Reply:
Warnings turn into errors in CI so the breakage should be the same.

Reply:
Ideally, this should have been a separate commit. So, fixing true source-code errors was separated from debatable build system config change.

Reply:
@pfalcon Agree. Maybe even another, later, PR?

Reply:
> Agree.

Thanks.

> Maybe even another, later, PR?

Whatever is easier for you. Just my 2 cents, I'd prefer it to be warning indeed, not error right away. As @SebastianBoe mentioned, warning will turn to errors during CI, so bad code won't leak into mainline by overlook. But during development/testing/debugging, sometimes it's helpful to take hacky shortcuts, so having it just as a warning is helpful.